url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/cmdlib/LoadLibrary.java

```diff
@@ -0,0 +1,251 @@
+            if(!file.canRead())
+                throw new CommandException(String.format("The file is not readable: \"%s\".", file.getAbsolutePath()));
+        }
+
+        public InputStream getStream()
+                throws CommandException
+        {
+            try
+            {
+                return new FileInputStream(file);
+            }
+            catch (FileNotFoundException e)
+            {
+                throw new CommandException(String.format("Exception while opening the file: '%s'.\n%s",file.getAbsolutePath(), e.getMessage()), e);
+            }
+        }
+
+        @Override
+        public String toString()
+        {
+            return file.getAbsolutePath();
+        }
+    }
+
+    public static class ClasspathLoader
+    implements Loader
+    {
+        String resource;
+
+        public ClasspathLoader(String aResource)
+        {
+            resource = aResource;
+        }
+
+        public void checkValidity() throws CommandException
+        {
+            if (this.getClass().getResourceAsStream(resource) == null)
+                throw new CommandException(String.format("The resource does not exist: \"%s\".", resource));
+        }
+
+        public InputStream getStream()
+                throws CommandException
+        {
+            return this.getClass().getResourceAsStream(resource);
+        }
+
+        @Override
+        public String toString()
+        {
+            return "classpath:" + resource;
+        }
+    }
+
+    @SuppressWarnings("unused")
+    public static class LoaderType
+    implements ITypeSpec
+    {
+        public String getSpecName()
+        {
+            return "loader";
+        }
+
+        public Object guard(Object aArg, IContext aCtx)
+        throws TypeSpecException
+        {
+            Loader lCandLdr;
+            if(aArg instanceof String)
+            {
+                String lPath = (String) aArg;
+                if(lPath.startsWith("classpath:"))
+                {
+                    lCandLdr = new ClasspathLoader(lPath.substring(10));
+                }
+                else if(lPath.startsWith("cp:"))
+                {
+                    lCandLdr = new ClasspathLoader(lPath.substring(3));
+                }
+                else
+                {
+                    // Interprete the string as a pathname.
+                    lCandLdr = new FileLoader(lPath);
+                }
+            }
+            else if (aArg instanceof File)
+            {
+                // Easy for us.
+                lCandLdr = new FileLoader((File) aArg);
+            }
+            else
+            {
+                // Don't know how to handle this.
+                throw new TypeSpecException(TypeUtil.msgBadRepr(getSpecName(), aArg.toString()));
+            }
+
+            try
+            {
+                lCandLdr.checkValidity();
+            }
+            catch (CommandException e)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)